Dropdown/DropdownList.js

```diff
@@ -0,0 +1,309 @@
+
+const getKey = ({children, selected}) => {
+	if (isSelectedValid({children, selected})) {
+		return children[selected].key;
+	}
+};
+
+const indexFromKey = (children, key) => {
+	let index = -1;
+	if (children) {
+		index = children.findIndex(child => child.key === key);
+	}
+
+	return index;
+};
+
+const DropdownListBase = kind({
+	name: 'DropdownListBase',
+
+	propTypes: {
+		/**
+		 * The selections for Dropdown
+		 *
+		 * @type {String[]|Array.<{key: (Number|String), children: (String|Component)}>}
+		 */
+		children: PropTypes.oneOfType([
+			PropTypes.arrayOf(PropTypes.string),
+			PropTypes.arrayOf(PropTypes.shape({
+				children: EnactPropTypes.renderable.isRequired,
+				key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
+			}))
+		]),
+
+		/**
+		 * Placement of the Dropdown List.
+		 *
+		 * @type {String}
+		 */
+		direction: PropTypes.string,
+
+		/**
+		 * Called when an item is selected.
+		 *
+		 * @type {Function}
+		 */
+		onSelect: PropTypes.func,
+
+		/**
+		 * Callback function that will receive the scroller's scrollTo() method
+		 *
+		 * @type {Function}
+		 */
+		scrollTo: PropTypes.func,
+
+		/**
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 */
+		selected: PropTypes.number,
+
+		/**
+		 * The current skin for this component.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		skin: PropTypes.string,
+
+		/**
+		 * The width of DropdownList.
+		 *
+		 * @type {('smallest'|'small'|'medium'|'large'|'x-large'|'huge')}
+		 */
+		width: PropTypes.oneOf(['smallest', 'small', 'medium', 'large', 'x-large', 'huge'])
+	},
+
+	defaultProps: {
+		direction: 'below center'
+	},
+
+	styles: {
+		css,
+		className: 'dropdownList'
+	},
+
+	handlers: {
+		itemRenderer: ({index, ...rest}, props) => {
+			const {children, selected, skin} = props;
+			const isSelected = index === selected;
+
+			let child = children[index];
+			if (typeof child === 'string') {
+				child = {children: child};
+			}
+			const data = child.children;
+			const ItemComponent = (skin === 'silicon') ? RadioItem : Item;
+			const itemProps = (skin === 'silicon') ? {className: css.dropDownListItem, css, selected: isSelected, size: 'small'} : {css, selected: isSelected};
+
+			return (
+				<ItemComponent
+					{...rest}
+					{...child}
+					data-selected={isSelected}
+					// eslint-disable-next-line react/jsx-no-bind
+					onClick={() => forward('onSelect', {data, selected: index}, props)}
+					{...itemProps}
+				/>
+			);
+		}
+	},
+
+	computed: {
+		className: ({direction, skin, width, styler}) => styler.append(direction.substr(0, direction.indexOf(' ')), width, {dropdownListWithCustomizedScroller: skin === 'silicon'}),
+		dataSize: ({children}) => children ? children.length : 0,
+		// Note: Retaining this in case we need to support different item sizes for large text mode:
+		itemSize: ({skin}) => (skin === 'gallium') ? ri.scale(90) : ri.scale(60)
```

Comment:
I changed this to like this since the other skins are all 60 except for gallium.